src/ai/backend/manager/models/keypair.py

```diff
@@ -148,6 +152,48 @@ def generate_ssh_keypair() -> Tuple[str, str]:
     return (public_key, private_key)
 
 
+def validate_ssh_keypair(
+    private_key_value: str, public_key_value: str
+) -> tuple[bool, Optional[str]]:
+    """
+    Validate RSA keypair for SSH/SFTP connection.
+
+    Args:
+        private_key_value: PEM-encoded private key string (OpenSSL format)
+        public_key_value: OpenSSH-encoded public key string
+
+    Returns:
+        tuple[bool, Optional[str]]:
+            Tuple containing a boolean indicating if the keypair is valid,
+            and an optional error message if invalid.
+    """
+
+    private_key = crypto_serialization.load_pem_private_key(
+        private_key_value.encode(),
+        password=None,  # No encryption as specified
+    )
+    public_key = crypto_serialization.load_ssh_public_key(public_key_value.encode())
+
+    # Check that both are RSA keys
+    if not isinstance(private_key, rsa.RSAPrivateKey) or not isinstance(
+        public_key, rsa.RSAPublicKey
+    ):
+        return False, "Keypair is not RSA"
```

Comment:
With so many algorithms like ECDSA and EDDSA, is it okay to just check RSA?

Reply:
Updated to cover ECDSA and ED25519 but not sure how many other algorithms to support